### PR TITLE
Add keyboard shortcuts for tab navigation

### LINKED ISSUE #2627: Add keyboard shortcuts for tab navigation
In most apps you can use ⌘1 to ⌘9 to switch from mayor group 1 to 9 (sometimes 9 means last and not 9th), and you can use ⇧⌘] and ⇧⌘[ to cycle forward and backwards through these groups (macOS shortcuts, Windows may use ⌃1 to ⌃9 etc).

In a browser or terminal app the major groups are tabs. In Rancher Desktop the major groups are the top level menu items on the left, so I would expect to use ⌘2 to switch to "Port Forwarding", ⌘3 to "Images" etc.

The same mechanism should work inside the preferences dialog as well, so if I type ⌘, followed by ⌘4 I should be seeing the "Kubernetes" preferences, and then pressing ⇧⌘] should cycle to the "Application" preferences.

The number key shortcuts are strictly based on position, and the corresponding tab may change if we insert additional menu items in the middle.

### PR DESCRIPTION
**Pull request by torchiaf**
Signed-off-by: Francesco Torchia <francesco.torchia@suse.com>

Fixes #2627

## TODO

- test key combos on macOS

### CODE REVIEW

File: pkg/rancher-desktop/window/preferences.ts [hunk 1]
```diff
@@ -62,6 +62,38 @@ export function openPreferences() {
...
+
+    Shortcuts.register(
+      window,
+      {
+        ...CommandOrControl,
+        key: '[',
+      },
+      () => window.webContents.send('route', { direction: 'back' }),
+      'switch preferences tabs by cycle [back]',
+    );
```
Comment (ericpromislow) [inline pkg/rancher-desktop/window/preferences.ts:96]: I realize it's not part of this PR, but it's the first time I've looked at this code, and the `if ... else` block at the end of `isRegistered` should be replaced with `return !!this.windows[id];`

**Review by ericpromislow — CHANGES_REQUESTED** [review]
Shouldn't the tab key work on the sub-panel navigation as well?

For example, when I'm on the `Application|General` pref panel, I can tab to the checkboxes and `Cancel` and `Apply`, but I can't tab to `Behavior` or `Environment`

File: pkg/rancher-desktop/window/preferences.ts (hunk 1 above)
Comment (torchiaf) [reply pkg/rancher-desktop/window/preferences.ts:96]: Fixed

**Review by ericpromislow — APPROVED** [review]
Looks ok, but I kept wanting to be able to use the arrow keys to navigate horizontally. Maybe a separate issue?

**Comment by torchiaf** [comment]
> Looks ok, but I kept wanting to be able to use the arrow keys to navigate horizontally. Maybe a separate issue?

Thanks Eric, I will bring this topic in our UI sync-up and open a separated issue in case.
